airbyte-integrations/connectors/source-linnworks/unit_tests/test_source.py

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+from unittest.mock import MagicMock
+
+import pytest
+from source_linnworks.source import LinnworksAuthenticator, SourceLinnworks
+
+
+@pytest.fixture
+def config():
+    return {"config": {"application_id": "xxx", "application_secret": "yyy", "token": "zzz", "start_date": "2021-11-01"}}
+
+
+@pytest.mark.parametrize(
+    ("response", "expected"),
+    [
+        (
+            {
+                "Token": "00000000-0000-0000-0000-000000000000",
+                "Server": "https://xx-ext.linnworks.net",
+            },
+            (True, None),
+        ),
+        (
+            {
+                "Code": None,
+                "Message": "Invalid application id or application secret",
+            },
+            (False, "Unable to connect to Linnworks API with the provided credentials - Error while refreshing access token: 'Token'"),
```

Comment:
```suggestion
            (False, "Unable to connect to Linnworks API with the provided credentials - Invalid application id or application secret"),
```

Reply:
The actual error message depends on which and how the authentication property is invalid. "Invalid application id or application secret" is one of few error messages. We should stick to the generic error message here or try to parse error response JSON and pick the value of the `Message` property.

Parsing of error response will add more complexity here. Is it worth it?

Reply:
If the API response provides more insight about the failed authentication I encourage you to display it to the user.